File: C#/1769 - CPF 1.cs
/// author Josélio de S. C. Júnior <joseliojrx25@gmail.com>
/// copyright Josélio de S. C. Júnior 2022

using System;
using System.Linq;
using System.Collections.Generic;
using System.Text.RegularExpressions;

class URI {
  static void Main() {
    while (true) {
      var gl = Console.ReadLine() ?? "";

      if (gl == "") break;

      int[] r = Regex.Replace(gl, @"\.|\-", "")
        .ToCharArray()
        .Select(e => Int32.Parse(e.ToString()))
        .ToArray();

      List<int> b1 = new List<int>(), b2 = new List<int>();

      for (int i = 0; i < 9; i++) {
        b1.Add(r[i] * (i + 1));
        b2.Add(r[i] * (9 - i));
      }

      int c1 = b1.Aggregate((a, b) => a + b) % 11;
      int c2 = b2.Aggregate((a, b) => a + b) % 11;
      c1 = c1 == 10 ? 0 : c1;
      c2 = c2 == 10 ? 0 : c2;

      Console.WriteLine(Int32.Parse(
          string.Concat(r.Skip(r.Length - 2).Take(2)))
          == Int32.Parse($"{c1}{c2}") ? "CPF valido"
          : "CPF invalido"
      );
    } 
  }
}

File: C/1769 - CPF 1.c
// author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
// copyright: Josélio Júnior (Lunatic Fox) 2023

#include "stdio.h"
#include "stdlib.h"

int main()
{
  char r[14];
  while (scanf("%s", &r) != EOF)
  {
    int b1 = 0, b2 = 0;
    int na[11];

    for (int i = 0, j = 0; i < 14; i++)
    {
      if (r[i] == '-' || r[i] == '.') continue;
      else 
      {
        char str[3] = {r[i], '\\', '0'};
        na[j] = atoi(str);
        j++;
      }
    }

    for (int i = 0, j = 9; i < 9; i++)
    {
      b1 += na[i] * (i + 1);
      b2 += na[i] * j;
      j--;
    }

    b1 %= 11; 
    b2 %= 11;

    b1 = b1 == 10 ? 0 : b1;
    b2 = b2 == 10 ? 0 : b2;

    if (b1 == na[9] && b2 == na[10])
      printf("CPF valido\n");
    else
      printf("CPF invalido\n");
  }
  return 0;
}
